pkg/etcd/etcd.go

```diff
@@ -1116,6 +1163,48 @@ func (e *ETCD) trackLearnerProgress(ctx context.Context, progress *learnerProgre
 	return e.setLearnerProgress(ctx, progress)
 }
 
+func (e *ETCD) setEtcdStatusCondition(node *v1.Node, promoted bool) (*v1.Node, error) {
+	etcdStatusType := v1.NodeConditionType("EtcdIsVoter")
+
+	var newCondition v1.NodeCondition
+	if promoted {
+		newCondition = v1.NodeCondition{
+			Type:    etcdStatusType,
+			Status:  "True",
+			Reason:  "MemberIsLearner",
+			Message: "Node is a voting member of the etcd cluster",
+		}
+	} else {
+		newCondition = v1.NodeCondition{
+			Type:    etcdStatusType,
+			Status:  "False",
+			Reason:  "MemberNotLearner",
+			Message: "Node has not been promoted to voting member of the etcd cluster",
+		}
+	}
+
+	for i, condition := range node.Status.Conditions {
+		if i == (len(node.Status.Conditions)-1) && condition.Type != etcdStatusType {
```

Comment:
Use this instead: https://github.com/kubernetes/kubernetes/blob/v1.28.3/pkg/controller/util/node/controller_utils.go#L292-L294